### PR TITLE
Feature/selection time indicator line

### PR DESCRIPTION
**Pull request by SudoPlz**
Allows to specify a component (amongst the other components) that will only get rendered, after the user clicks, but before they start dragging.

Example:

```
components={{
    beforeSelectIndicatorComponent: (selectionProps) => <BigCalendarPreSelectIndicator {...selectionProps} />,
}}
```

Now I can do this after clicking:

See the dashed line and the time tool tip before dragging?

### CODE REVIEW

File: src/DayColumn.js [hunk 1]
```diff
@@ -77,6 +77,44 @@ class DayColumn extends React.Component {
...
+      let selectDates = { start: startDate, end: endDate }
+
+      let selectionElement
+      if (
+        noMovementYet &&
+        components &&
+        components.beforeSelectIndicatorComponent
+      ) {
+        // if the user is about to start selecting but hasn't started dragging yet
+        selectionElement = components.beforeSelectIndicatorComponent({
```
Comment (jquense) [inline src/DayColumn.js]: this should be a real component not a render prop to be consistent with the other apis

File: src/DayColumn.js [hunk 2]
```diff
@@ -275,9 +308,17 @@ class DayColumn extends React.Component {
     selector.on('selectStart', maybeSelect)
 
     selector.on('beforeSelect', box => {
+      const didNotClickOnAnEvent = !isEvent(findDOMNode(this), box)
+      if (
+        didNotClickOnAnEvent &&
+        this.props.components &&
+        this.props.components.beforeSelectIndicatorComponent
+      ) {
+        maybeSelect({ ...box, noMovementYet: true })
```
Comment (jquense) [inline src/DayColumn.js]: there shouldn't be side effects in beforeSelect if possible

File: src/addons/dragAndDrop/withDragAndDrop.js [hunk 3]
```diff
@@ -136,10 +136,10 @@ export default function withDragAndDrop(Calendar) {
       )
 
       props.components = {
-        ...components,
         eventWrapper: EventWrapper,
         eventContainerWrapper: EventContainerWrapper,
         weekWrapper: WeekWrapper,
+        ...components,
```
Comment (jquense) [inline src/addons/dragAndDrop/withDragAndDrop.js]: this is handled on master now already

File: src/DayColumn.js (hunk 1 above)
Comment (SudoPlz) [reply src/DayColumn.js]: Roger that, I'll make the change.

File: src/DayColumn.js (hunk 2 above)
Comment (SudoPlz) [reply src/DayColumn.js]: Not sure I understand, can you explain a bit better what you mean by that?

File: src/addons/dragAndDrop/withDragAndDrop.js (hunk 3 above)
Comment (SudoPlz) [reply src/addons/dragAndDrop/withDragAndDrop.js]: got it.

File: src/DayColumn.js (hunk 2 above)
Comment (jquense) [reply src/DayColumn.js]: the idea of this callback is only to determine whether the selection should be handled, it should do any work, like storing coordinates. the `mousedown`, `click`, `selectStart`, etc handlers are for that

Comment (SudoPlz) [reply src/DayColumn.js]: Got it, so I can do something like:

```
selector.on('mousedown', box => {
      const didNotClickOnAnEvent = !isEvent(findDOMNode(this), box)
      if (
        didNotClickOnAnEvent &&
        this.props.components &&
        this.props.components.beforeSelectIndicatorComponent != null
      ) {
        maybeSelect({ ...box, noMovementYet: true })
      }
    })
```
?
I'll try that.

Comment (SudoPlz) [reply src/DayColumn.js]: Hmm, nope that doesn't work. I need that code to run on mouse down, but seems like `selector` never receives that event.

File: src/DayColumn.js [hunk 4]
```diff
@@ -77,6 +77,34 @@ class DayColumn extends React.Component {
...
+      let { top, height, startDate, endDate, noMovementYet } = this.state
+      const { localizer, components } = this.props
+      let selectDates = { start: startDate, end: endDate }
+      let selectionElement
+      if (
+        noMovementYet &&
+        components &&
+        components.beforeSelectIndicatorComponent != null
+      ) {
+        const SelectIndicator = components.beforeSelectIndicatorComponent
```
Comment (jquense) [inline src/DayColumn.js]: sorry so delayed here :(

Can we just skip the component and render something like your screenshot as the default? Then folks can customize it via css instead

Comment (SudoPlz) [reply src/DayColumn.js]: Sure we can do that, I'll revise my PR when I find some free time and post back.

Comment (SudoPlz) [reply src/DayColumn.js]: https://github.com/intljusticemission/react-big-calendar/pull/1018 is ready

**Comment by SudoPlz** [comment]
Closing this in favour of https://github.com/intljusticemission/react-big-calendar/pull/1018

**Pull request closed without merging**